trulens_eval/trulens_eval/tru.py

```diff
@@ -163,6 +168,8 @@ def __init__(
         database_prefix: Optional[str] = None,
         database_args: Optional[Dict[str, Any]] = None,
         database_check_revision: bool = True,
+        snowflake_connection_parameters: Optional[Dict[str, str]] = None,
```

Comment:
+1, lets make this a Pydantic model?

Reply:
Is there some canonical snowflake connection object? I wasn't aware of one when I wrote this.

I did it this way because that's the way that Snowpark does it. See here:
https://docs.snowflake.com/en/developer-guide/snowpark/python/creating-session#connect-by-specifying-connection-parameters
I don't really want think we should do the `.toml` approach (i.e. the snowsql approach) for the reasons I listed in the description of the PR so that only really leaves this approach in that doc.

Reply:
I was thinking of this: https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-api#object-connection

Don't think we should read params from `.toml` but if users specify a connection object with their own params through snowpark I think we should adopt that standard

Reply:
I think this is better. Dave's description covers this.

Reply:
@coreyhu, unfortunately AFAICT, with that object I can't get the password out which I need to connect to Snowflake via sqlalchemy.